Tests/TSCBasicTests/FileSystemTests.swift

```diff
@@ -73,13 +73,45 @@ class FileSystemTests: XCTestCase {
                     _ = try fs.getDirectoryContents(AbsolutePath("/does-not-exist"))
                     XCTFail("Unexpected success")
                 } catch {
-                    XCTAssertEqual(error.localizedDescription, "The folder “does-not-exist” doesn’t exist.")
+                    XCTAssertEqual(error._domain, "NSCocoaErrorDomain")
+                    XCTAssertEqual(error._code, 260)
+                    XCTAssert(error.localizedDescription.contains("does-not-exist"))
+                    // English description: “The folder “does-not-exist” doesn’t exist.”
                 }
 
                 let thisDirectoryContents = try! fs.getDirectoryContents(AbsolutePath(#file).parentDirectory)
                 XCTAssertTrue(!thisDirectoryContents.contains(where: { $0 == "." }))
                 XCTAssertTrue(!thisDirectoryContents.contains(where: { $0 == ".." }))
                 XCTAssertTrue(thisDirectoryContents.contains(where: { $0 == AbsolutePath(#file).basename }))
+
+                // Unicode
+                let willyNillyString = "e\u{301}\u{E9}x\u{304}\u{331}" // Neither NFC nor NFD.
+                let nfdString = "e\u{301}e\u{301}x\u{331}\u{304}"
+                let nfcString = "\u{E9}\u{E9}x\u{331}\u{304}"
+                #if !os(Linux) // TODO: Linux cannot normalize?!?
+                XCTAssertEqual(
+                    Array(willyNillyString.decomposedStringWithCanonicalMapping.unicodeScalars),
+                    Array(nfdString.unicodeScalars))
+                XCTAssertEqual(
+                    Array(willyNillyString.precomposedStringWithCanonicalMapping.unicodeScalars),
+                    Array(nfcString.unicodeScalars))
+                #endif
+                let willyNilly = tempDirPath.appending(component: willyNillyString)
+                let nfd = tempDirPath.appending(component: nfdString)
+                let nfc = tempDirPath.appending(component: nfcString)
+                let variants = [willyNilly, nfd, nfc]
+                for directory in variants {
+                    do {
+                        try fs.createDirectory(directory)
+                        defer { try? fs.removeFileTree(directory) }
+                        for other in variants {
+                            _ = try fs.getDirectoryContents(other)
+                        }
+                    } catch {
+                        let directoryName = directory.basename.unicodeScalars.map({ "U+\($0.value)" })
+                        XCTFail("Unable to handle directory named “\(directoryName)”: \(error)")
+                    }
+                }
             }
         }
     }
```

Comment:
If you're looking for a more complete programmatic data set to drive this, the stdlib uses Unicode's published normalization tests at https://raw.githubusercontent.com/apple/swift/master/test/stdlib/Inputs/NormalizationTest.txt

Reply:
This lightweight test is really only there to check that Unicode equivalence is being considered at all. `String` does all the heavy lifting when it comes to the definition of equivalence. ([The text fixture we are trying to get working on Linux](https://github.com/apple/swift-package-manager/blob/209d3af0487c78ab20009818bab9e305fc9d62a1/Fixtures/Miscellaneous/Unicode/Package.swift#L6-L17) is already much meaner too.)

Reply:
You can also try `U+0F73` which is FCD, has a CCC of 0, and yet expands into two different scalars, neither of which have a CCC of 0.